src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -193,6 +193,11 @@ public void Serialize(ObjectWriter writer, CancellationToken cancellationToken)
                     continue;
                 }
 
+                if (Equals(value, optionKey.Option.DefaultValue))
```

Comment:
nice.  possibly worth a comment.

Reply:
@mavasani @jasonmalinowski Is this a valid optimization? If so, it's one of the bigger wins in this pull request.

Reply:
Looks reasonable to me - from quick glance at the uses of `_serializableOptionValues`, it doesn't seem checksum validation will be affected.

Reply:
This one is causing test failures. I'm going to back it out so we can get the rest in.